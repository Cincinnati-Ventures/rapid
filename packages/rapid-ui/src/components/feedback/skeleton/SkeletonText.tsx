import React from 'react';
import { RapidStyles } from '../../../utils';
import classNames from '../../../conditional';
import { twMerge } from 'tailwind-merge';
import { VStack } from '../../layout';
import type { Spacing } from '../../../types';
import type { Speed } from './';

const RAPID_CLASSNAME = 'rapid-skeleton';

interface SkeletonProps extends React.HTMLAttributes<HTMLDivElement> {
	styles?: string;
	isLoading?: boolean;
    containerStyles?: string;
    lineSpacing?: Spacing;
    numberOfLines?: number;
    speed?: Speed;
}

const SkeletonText = React.forwardRef<HTMLDivElement, SkeletonProps>(
	({ styles, containerStyles, lineSpacing, isLoading = true, numberOfLines = 4, speed }, ref) => {
		const defaultStyles = 'h-3 w-40 rounded-md';
		return (
            <VStack styles={containerStyles} spacing={lineSpacing}>
                {Array.from(Array(numberOfLines).keys()).map((number, index) => {
                    return (
                        <div
                            key={index + number}
                            ref={ref}
                            className={twMerge(
                                RapidStyles(
                                    styles,
                                    defaultStyles,
                                    RAPID_CLASSNAME,
                                ),
                                classNames({
                                    condition: isLoading,
                                    classNames: speed || 'skeleton-pulse',
                                }),
                            )}
                        />
                    )
                })}
            </VStack>
		);
	},
);

SkeletonText.displayName = "SkeletonText";

export default SkeletonText;
