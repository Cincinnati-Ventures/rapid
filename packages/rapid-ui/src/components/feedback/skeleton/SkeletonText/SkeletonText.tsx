import React from 'react';
import { RapidStyles } from '../../../../utils';
import classNames from '../../../../conditional';
import { twMerge } from 'tailwind-merge';
import { VStack } from '../../../layout';
import type { Spacing } from '../../../../types';
import type { Speed } from '..';

interface SkeletonTextProps extends React.HTMLAttributes<HTMLDivElement> {
	isLoading?: boolean;
	numberOfLines?: number;
	speed?: Speed;
	lineSpacing?: Spacing;
	containerStyles?: string;
	styles?: string;
}

const RAPID_CLASSNAME = 'rapid-skeleton';
const SKELETON_TEXT_STYLES = 'h-3 w-full rounded-md';

const SkeletonText = React.forwardRef<HTMLDivElement, SkeletonTextProps>(
	(
		{
			styles,
			containerStyles,
			lineSpacing,
			isLoading = true,
			numberOfLines = 4,
			speed,
		},
		ref,
	) => {
		return (
			<VStack styles={containerStyles} spacing={lineSpacing}>
				{Array.from(Array(numberOfLines).keys()).map(
					(number, index) => {
						return (
							<div
								key={index + number}
								ref={ref}
								className={twMerge(
									RapidStyles(
										styles,
										SKELETON_TEXT_STYLES,
										RAPID_CLASSNAME,
									),
									classNames({
										condition: isLoading,
										classNames: speed || 'skeleton-pulse',
									}),
								)}
							/>
						);
					},
				)}
			</VStack>
		);
	},
);

SkeletonText.displayName = 'SkeletonText';

export default SkeletonText;
