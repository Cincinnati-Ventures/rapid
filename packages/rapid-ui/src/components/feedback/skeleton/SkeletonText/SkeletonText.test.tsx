import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import SkeletonText from './SkeletonText';

describe('SkeletonText', () => {
	it('should render default correctly', () => {
		const { container } = render(<SkeletonText />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with props', () => {
		const { container } = render(
			<SkeletonText
				styles='custom-class'
				containerStyles='custom-container-class'
				lineSpacing='lg'
				isLoading={false}
				speed='skeleton-pulse-fast'
			/>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with a couple props', () => {
		const { container } = render(
			<SkeletonText
				isLoading
				lineSpacing='sm'
				speed='skeleton-pulse-medium'
			/>,
		);
		expect(container).toMatchSnapshot();
	});
});
