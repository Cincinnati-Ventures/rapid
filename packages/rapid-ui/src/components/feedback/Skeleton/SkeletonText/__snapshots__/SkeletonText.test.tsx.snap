// Vitest Snapshot v1

exports[`SkeletonText > should render correctly with a couple props 1`] = `
<div>
  <div
    class="flex flex-col space-y-1.5 rapid-stack"
  >
    <div
      class="h-3 w-full rounded-md rapid-skeleton skeleton-pulse-medium"
    />
    <div
      class="h-3 w-full rounded-md rapid-skeleton skeleton-pulse-medium"
    />
    <div
      class="h-3 w-full rounded-md rapid-skeleton skeleton-pulse-medium"
    />
    <div
      class="h-3 w-full rounded-md rapid-skeleton skeleton-pulse-medium"
    />
  </div>
</div>
`;

exports[`SkeletonText > should render correctly with props 1`] = `
<div>
  <div
    class="flex flex-col space-y-6 custom-container-class rapid-stack"
  >
    <div
      class="h-3 w-full rounded-md custom-class rapid-skeleton"
    />
    <div
      class="h-3 w-full rounded-md custom-class rapid-skeleton"
    />
    <div
      class="h-3 w-full rounded-md custom-class rapid-skeleton"
    />
    <div
      class="h-3 w-full rounded-md custom-class rapid-skeleton"
    />
  </div>
</div>
`;

exports[`SkeletonText > should render default correctly 1`] = `
<div>
  <div
    class="flex flex-col space-y-6 rapid-stack"
  >
    <div
      class="h-3 w-full rounded-md rapid-skeleton skeleton-pulse"
    />
    <div
      class="h-3 w-full rounded-md rapid-skeleton skeleton-pulse"
    />
    <div
      class="h-3 w-full rounded-md rapid-skeleton skeleton-pulse"
    />
    <div
      class="h-3 w-full rounded-md rapid-skeleton skeleton-pulse"
    />
  </div>
</div>
`;
